POEM_071.md

```diff
@@ -0,0 +1,41 @@
+POEM ID: 071  
+Title: POEM_071 - Change ExecComp to use `declare_coloring`  
+authors: @robfalck  
+Competing POEMs:  
+Related POEMs:  
+Associated implementation PR: N/A  
+
+Status:
+
+- [ ] Active
+- [ ] Requesting decision
+- [x] Accepted
+- [ ] Rejected
+- [ ] Integrated
+
+
+## Motivation
+
+The `ExecComp` in OpenMDAO uses complex-safe functions to provide user-defined calculations within a run script in leiu of writing a custom component.
+
+The option `has_diag_partials` on ExecComp was added prior to the development of partial derivative coloring to provide a common sparsity pattern for vectorized components.
+
+With the advent of partial coloring, `has_diag_partials` is no longer needed.
+Users should be able to expect that `ExecComp` will determine the sparsity pattern of calualtions involved.
+
+## Changes
+
+1. `has_diag_partials` will be marked as deprecated but still be allowed as an option that does nothing.
```

Comment:
We ended up leaving the old internal CS stuff intact because that allows us to use CS local to the ExecComp without requiring the full model vectors to be allocated as complex, and has_diag_partials offers a performance improvement over computing the coloring so I left it in (with no deprecation).